org.pjbroad.EternallandsClient.json

```diff
@@ -0,0 +1,169 @@
+{
+	"app-id": "org.pjbroad.EternallandsClient",
+	"runtime": "org.freedesktop.Platform",
+	"sdk": "org.freedesktop.Sdk",
+	"runtime-version": "1.6",
+	"command": "launcher.sh",
+	"finish-args": [
+		"--share=ipc",
+		"--socket=x11",
+		"--socket=wayland",
+		"--socket=pulseaudio",
+		"--share=network",
+		"--device=dri",
+		"--filesystem=~/.elc:create"
```

Comment:
You can use `--persist=.elc`.

Reply:
(This means that the .elc directory will be created in `~/.var/app/org.pjbroad.EternallandsClient/` instead of `~/`.)

Reply:
I must admit that I did not fully understand the description in the docs about what --persist is for.  What I want is to be able to use ~/.elc if it exists and create ~/.elc if is does not already exist. Also, for changes in ~/.elc to be persistent and unaffected if the package is uninstalled.  The reason for is is to provide compatibility with existing installs of the game that do not use flatpak and to keep the "standard" location these user files the same so others can help easily with issues.  It looks like "--filesystem=~/.elc:create" and "--persist=~/.elc:create" both do this but I do not know which is best or what they do differently.  I'd appreciate your advise on this.

Reply:
The directory is being created by `--persist=.elc` too if missing; none of the files created by applications are removed on uninstall. Note the path is already homedir relative.

It binds '~/.elc` in the sandbox to '~/.var/app/appid/.elc` so it won't migrate the already existing data.

Reply:
Thanks for the further explanation, but I still don't get it 100%, the need for mapping in particular.  I need the user files to be in the ~/.elc/ for all the reasons above.  Either already there or created there.  I've tried experimented with "--persist=\~/.elc" but that does not create the directory if missing "--persist=\~/.elc:create" does create it so I'll use that.  I'll remove the leading ~ but it looks more obvious what's happening with it present I'd suggest.

Reply:
So I've done a bunch more testing, including removing the files that get stored in ~/.var/app to make sure its all celan.  I was wrong, --persist does not work, ~/elc/ path is not created.  Therefore I'm sticking with --filesystem which does work.  Hope that's OK.

Reply:
I'll try to explain it better (sorry if I fail!) The diference between --filesystem is that application sees it as is; so if you pass --filesystem=~/foo, it will be visible from app perspective as ~/foo and it will be stored outside the sandbox also as ~/foo. Persist is a hack to make applications not following XDG spec see their directory at correct location for them (so, `~/.eic` here; only in the sandbox!) but in reality stored elsewhere outside the sandbox, in this instance `~/.var/app/org.pjbroad.EternallandsClient/.eic`. This is why you don't see it created – it is being created from your not-in-sandbox point of view but elsewhere.

I'd much prefer if we could keep `--persist` but of course I can't force you if that's your decision.

Reply:
I think I finally get it, many thanks for taking more time to explain and I can see why that approach is useful / preferred.  However, for this application, user will likely interact with their config files stored in ~/.elc/, creating new player configs and changing elements of the game UI etc.  Existing players of the game switching to flatpak will want their existing config files to be used.  If they choose to switch back to some other packages version or one they have built themselves, they will want to keep using the same config files.  If we use --persist the user will have to manage this transition themselves.  Also, from a support perspective, using the same location across all Linux versions is desirable.  Hopefully I have understood now and have suitably explained my reasoning.